executor/explain.go

```diff
@@ -89,6 +99,16 @@ func (e *ExplainExec) executeAnalyzeExec(ctx context.Context) (err error) {
 				}
 			}
 		}()
+		if e.ctx.GetSessionVars().MemoryDebugMode > 0 {
+			exit := make(chan bool)
+			e.runMemoryDebugGoroutine(exit)
+			defer func() {
+				select {
+				case <-exit:
+				}
+			}()
+			defer func() { exit <- false }()
```

Comment:
This is weird. How does it exit? Can't we use waitgroup to wait the service go routine?